### DISCUSSION TITLE
Add new pattern for Markdown code blocks

### DISCUSSION THREAD

**Discussion by AlightYoung** [markedjs/marked#2817] [Q&A]
Currently I need to continuously render a streaming markdown content. 

However, when rendering a large piece of code block content (in the code block content, but not yet finished.), the rendering will sometimes become a code block and sometimes ordinary text, which makes the structure of the webpage constantly adjusting. The visual result is constant flickering.

So I want to ensure that the structure will not change intermittently by defining the following format as a code block. (I also don't know if this avoids the above problems.)

```
```html
<body>
...
</body>
```

Any help is appreciated.

**Comment by UziTech** [comment]
You can create a [tokenizer](https://marked.js.org/using_pro#tokenizer) that does this.

**Reply by AlightYoung** [reply]
Thank you. I have reread the doc and rewritten a tokenizer like this: 

```javascript
const tokenizer = {
  code(src) {
    const match = src.match(/^`{3}([\S]+)?\n([\s\S]+)/)
    if (match) {
      return {
        type: 'code',
        raw: match[0],
        text: match[1].trim()
      }
    }
    return false
  }
}
```

But it doesn't work. Is there something in the doc that I overlooked?

**Reply by UziTech** [reply]
Can you give an example of what your input is and what you expect for the output?

**Reply by AlightYoung** [reply]
After a new round of testing, I found that rendering a large number of code blocks does not have the above problems.

Maybe the problem is just me using React's UseEffect hook (or another things) incorrectly. I'm ready to refactor my code.

Anyway thanks a lot.
